Applications/Utils/ModelPreparation/PartitionMesh/MeshPartitioning.h

```diff
@@ -0,0 +1,86 @@
+/*!
+  \file MeshPartitioning.h
+  \date   2016.05
+
+  \brief  Declare a class to perform mesh partitioning
+
+  \copyright
+  Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+             Distributed under a Modified BSD License.
+               See accompanying file LICENSE.txt or
+               http://www.opengeosys.org/project/license
+
+*/
+
+#ifndef MESH_PARTITIONING_H_
+#define MESH_PARTITIONING_H_
+
+#include <vector>
+#include <string>
+#include <fstream>
+
+#include "MeshLib/Mesh.h"
+
+namespace MeshLib
+{
+
+enum class ElementType : unsigned {LINE2, QUAD4, HEX8, TRI3, TET4, PRISM6, PYRAMID5, INVALID};
+
+/// A subdomain mesh.
+class MeshPartitioning : public Mesh
+{
+        typedef long PetscInt;
+    public:
+        /// Copy constructor
+        MeshPartitioning(const MeshLib::Mesh &mesh) : Mesh(mesh)
```

Comment:
Actually, that's not exactly a copy ctor. It's a conversion operator. Please make it `explicit` and change the comment, maybe to:
`/// Construct an instance from a given \c Mesh`.